interfaces/builtin/steam_support.go

```diff
@@ -179,6 +182,8 @@ pivot_root oldroot=/newroot/ /newroot/,
 umount /,
 
 # Permissions needed within sandbox root
+/usr/bin/** ixr,
```

Comment:
This should probably have a short comment, i.e. that this does *not* make the coreXX /{,s}bin available but the PV generated one

Reply:
Unfortunately, it does make executables from the base snap available too. We're not triggering a profile transition on entering the PV sandbox, so the same AppArmor rules apply in both contexts. And right now it is only Windows games that default to running in a PV sandbox.

I'm not sure what the best solution for this is. In theory, access to the executables should be a security issue as they'll run with the same permissions as the snap. There is the compatibility concern that we don't want to promise that all the binaries present in the bases will remain so, but I think that is somewhat offset by the fact that the games expect to run on any random Linux. So they have even less guarantee of what executables will be available than a snap would.

Reply:
Whilst an argument can be made that there is a security concern here, since a snap could just stage and ship whatever binaries it wants anyway, this really just increases the possible attack surface against a snap since it now has a lot more binaries (and hence code) that could contain various vulnerabilties.

However, the only thing that may give some cause for concern is set-uid root executables - in particular this would grant access to `/usr/bin/sudo` and `/bin/su` which are both setuid-root.

I wonder if there would be value in adding explicit `deny` rules for these (assuming PV doesn't actually need access to these)?

Reply:
I've added the following:

```
deny /usr/bin/{chfn,chsh,gpasswd,mount,newgrp,passwd,su,sudo,umount} x,
```

Which covers the setuid binaries in `/usr/bin` on core20 and core22. I doubt anything inside PV's sandbox could make use of these executables even if they are present:

1. It downloads the Steam runtime as a regular user, so couldn't make them setuid root.
2. Bubblewrap configures most of its mount namespace with nosuid
3. Bubblewrap calls `prctl (PR_SET_NO_NEW_PRIVS, 1, ...)`, so the game couldn't successfully call a setuid binary even if it was available.

Reply:
Thanks - this list of binaries matches with what we have in review-tools as well (https://git.launchpad.net/review-tools/tree/reviewtools/overrides.py#n292) - also thanks for the extra info re bubblewrap - it sounds like it all should be as safe as we can make it at this point.